features/networking/ovn_ipsec.feature

```diff
@@ -0,0 +1,158 @@
+    Given I store all worker nodes to the :workers clipboard
+    #Getting ovn-ipsec pod name for a corresponding worker node in sight
+    When I run the :get admin command with:
+      | resource      | pod                                   |
+      | l             | app=ovn-ipsec    |
+      | fieldSelector | spec.nodeName=<%= cb.workers[0].name %> |
+      | n             | openshift-ovn-kubernetes              |
+      | o             | jsonpath={.items[*].metadata.name}    |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :ovn_ipsec_pod clipboard
+    Given I use the "<%= cb.workers[0].name %>" node
+    #Simulating segfault on pluto IKE dameon
+    And I run commands on the host:
+      | pgrep -SEGV pluto|
+    Then the step should succeed
+    #Need to give it some hard coded time for ovn-ipsec pod to notice segfault
+    Given 90 seconds have passed
+    #Checking readiness probe functionality to make sure it re-creates corresponding ovn-ipsec pod
+    And admin waits for all pods in the "openshift-ovn-kubernetes" project to become ready up to 300 seconds
+    
+  # @author anusaxen@redhat.com
+  # @case_id OCP-37591
+  @admin
+  Scenario: Make sure IPSEC-SA's are establishing in a transport mode
+    Given the env is using "OVNKubernetes" networkType
+    And the IPSec is enabled on the cluster
+    Given I select a random node's host
+    And I run commands on the host:
+      | grep -i "IPsec SA established transport mode" /var/log/openvswitch/libreswan.log |
+    Then the step should succeed
+    #We need to make sure some mode is chosen and supported only is trasport
+    And the output should contain "IPsec SA established transport mode"
+    
+  # @author anusaxen@redhat.com
+  # @case_id OCP-39216
+  @admin
+  Scenario: Pod created on IPSec cluster should have appropriate MTU size to accomdate IPSec Header
+    Given the env is using "OVNKubernetes" networkType
+    And the IPSec is enabled on the cluster
+    Given the default interface on nodes is stored in the :default_interface clipboard
+    And the node's MTU value is stored in the :cluster_mtu clipboard
+    Given I have a project
+    And I have a pod-for-ping in the project
+    When I execute on the pod:
+      | bash | -c | cat /sys/class/net/eth0/mtu |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :test_pod_mtu clipboard
+    # OVN needs 100 byte header and IPSec needs another 46 bytes due to ESP etc so the pod's mtu must be 146 bytes less than cluster mtu
+    And the expression should be true> cb.test_pod_mtu.to_i + 146 == cb.cluster_mtu.to_i
+ 
+  # @author anusaxen@redhat.com
+  # @case_id OCP-37590
+  @admin
+  @destructive	
+  Scenario: Delete all ovn-ipsec containers and check if they gets recreated
+    Given the env is using "OVNKubernetes" networkType
+    And the IPSec is enabled on the cluster
+    When I run the :delete admin command with:
```

Comment:
are these pods backed by an operator that's responsible to make sure they are recreated?

Reply:
Yep CNO monitors that